src/index.ts

```diff
@@ -44,117 +44,7 @@ import {
   TranslationBundle
 } from '@jupyterlab/translation';
 import { ISettingRegistry } from '@jupyterlab/settingregistry';
-import { YFile } from '@jupyterlab/shared-models';
-
-export class GalyleoModel
-  extends CodeEditor.Model
-  implements DocumentRegistry.ICodeModel
```

Comment:
As you said you actually don't need a specific model, you are simply applying your viewer on a file with a specific extension. Therefore you don't need a custom model,...

Reply:
That would be wonderful (as we both know, the less code, the better :-)).  But without a `Model` and an associated `ModelFactory`, I don't know how to associate an open command with a specific editor, since (AFAIK) the `DocumentRegistry` operates on models, not extensions.   But you are perfectly right, there should be no code in the `GalyleoModel` class, and the only code in the `GalyleoModelFactory` class should be a `createNew` method, which just returns a `GalyleoModel`.
If there is a better way to do this, (I am far from an expert), I would love to do it.

Reply:
Ah, I looked at the code again, and now I see what you mean!  Thanks